### PR TITLE
feat: big query importer

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### Issue

upserting tables from big query into core

[[Link to Basecamp Todo](https://3.basecamp.com/3105655/projects)](https://3.basecamp.com/3105655/buckets/37071156/todos/7356102081)

### CODE REVIEW

File: apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts [hunk 1]
```diff
@@ -23,27 +25,52 @@ const videoSchema = object({
...
       where: { id: video.id }
     })
-    if (record != null)
+    if (record != null) {
+      const videoObj = await this.transform(video)
       await this.prismaService.video.update({
-        where: { id: video.id },
-        data: video
+        where: { id: videoObj.id },
+        data: videoObj
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts:72]:
```suggestion
        data: {
          ...video,
          slug: this.slugify(video.id, video.slug)
        }
```

File: apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts [hunk 2]
```diff
@@ -23,27 +25,52 @@ const videoSchema = object({
...
+export class ImporterVideosService extends ImporterService<Video> {
   schema = videoSchema
+  usedSlugs: Record<string, string> | undefined
+  ids: string[] = []
 
   constructor(private readonly prismaService: PrismaService) {
     super()
   }
 
+  private async transform(video: Video): Promise<Prisma.VideoCreateInput> {
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts:44]: change name to slugify and return a string that = slug. Args should be id, title.

**Review by tataihono — CHANGES_REQUESTED** [review]

File: apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts [hunk 3]
```diff
@@ -23,27 +25,44 @@ const videoSchema = object({
...
+  }
+
   protected async save(video: Video): Promise<void> {
-    const record = await this.prismaService.video.findUnique({
-      where: { id: video.id }
+    await this.prismaService.video.upsert({
+      where: { id: video.id },
+      update: { ...video, slug: await this.slugify(video.id, video.slug) },
+      create: { ...video, slug: await this.slugify(video.id, video.slug) }
     })
```
Comment (tataihono) [inline apps/api-videos/src/app/modules/importer/importerVideos/importerVideos.service.ts:66]:
```suggestion
    const input = { ...video, slug: await this.slugify(video.id, video.slug) }
    await this.prismaService.video.upsert({
      where: { id: video.id },
      update: input,
      create: input
    })
```

**Review by tataihono — CHANGES_REQUESTED** [review]
please drop your tests from this. Do the compare on a separate branch. @mikeallisonJS can you help @Kneesal compare the seed import to the bq import here?

**Comment by Kneesal** [comment]
this has been re-assigned to @mikeallisonJS  who will complete this in a separate PR

**Pull request closed without merging**